app/geocode/DynamicMarker.tsx

```diff
@@ -19,16 +19,20 @@ function DynamicMarker({
     useState<string>("");
   const map = useMap();
 
-  const getGeoAsNumber = (lat: boolean, long: boolean): number | null => {
-    const columnName = lat ? latColumnName : long ? longColumnName : "";
-    if (!record || !record[columnName]) {
-      return null;
-    }
-    // Grist in French version use "," as decimal separator, it could break geo api
-    return typeof record[columnName] === "string" && record[columnName].includes(",") ?
-      Number(record[columnName].replace(",", ".")) :
-      Number(record[columnName]);
-  }
+  const getGeoAsNumber = useCallback(
+    (lat: boolean, long: boolean): number | null => {
+      const columnName = lat ? latColumnName : long ? longColumnName : "";
+      if (!record || !record[columnName]) {
+        return null;
+      }
+      // Grist in French version use "," as decimal separator, it could break geo api
+      return typeof record[columnName] === "string" &&
+        (record[columnName]! as string).includes(",")
+        ? Number((record[columnName]! as string).replace(",", "."))
+        : Number(record[columnName]);
```

Comment:
mmh I don't know ... I needed to do that for the linter which didn't understand that it could be only a string and nothing else so it just for the linter to understand is a string, we don't have to convert it.

Reply:
My proposal is to replace the whole `return` statement, without the ternary expression.

> I needed to do that for the linter which didn't understand that it could be only a string and nothing else

So in fact the value can only be a string? I maintain my proposal in any case, `toString()` is just a way to be sure we manipulate a string and help the compiler admit that. I expect the conversion of a string to the same type to have no cost (computation or memory allocation), or to be insignificant. 

Here I rather worry about the expressiveness and the complexity.

Reply:
https://github.com/betagouv/grist-custom-widgets-fr-admin/pull/35